### PR TITLE
Instructor solutions update

### PR DESCRIPTION
**Pull request by PBarmby**
This replaces PR #99. I think all of the changes requested by @abostroem have been made. Since the discussion about where to put the solutions is still ongoing (see [lesson-template#267](https://github.com/swcarpentry/lesson-template/issues/267)) I haven't done anything about that.

I hope to get to the solutions for 10-cmdline at some point.

### CODE REVIEW

File: instructors.md [hunk 1]
```diff
@@ -90,18 +90,548 @@ which are covered in the lesson on the shell.
...
+> ~~~ {.output}
+> newstring = ''
+> oldstring = 'Newton'
+> length_old = len(oldstring)
+> for c in range(length_old):
+>    newstring = newstring + oldstring[length_old-c-1]
+> print newstring
+> 'notweN'
+> ~~~
+
```
Comment (abostroem) [inline instructors.md]: Can we use something more descriptive than c as the variable in the loop to demonstrate how we would like our students to program?

File: instructors.md [hunk 2]
```diff
@@ -90,18 +90,548 @@ which are covered in the lesson on the shell.
...
+> f = 0
+> k = 0
+>
+> def f2k(f):
+>   k = ((f-32)*(5.0/9.0)) + 273.15
+>   return k
+>
+> f2k(8)
+> f2k(41)
+> f2k(32)
```
Comment (abostroem) [inline instructors.md]: We probably want print statements in front of the f2k for those people doing this in a single cell in the notebook

File: instructors.md [hunk 3]
```diff
@@ -90,18 +90,548 @@ which are covered in the lesson on the shell.
...
+> 1. Read the code below, and (without running it) try to identify what the errors are.
+> 2. Run the code, and read the error message. Is it a `SyntaxError` or an `IndentationError`?
+> 3. Fix the error.
+> 4. Repeat steps 2 and 3, until you have fixed all the errors.
+>
+> ~~~ {.python}
+> def another_function
+>   print "Syntax errors are annoying."
+>    print "But at least python tells us about them!"
+>   print "So they are usually not too hard to fix."
```
Comment (abostroem) [inline instructors.md]: It looks like something is funny with the indentation here...

File: instructors.md [hunk 4]
```diff
@@ -90,18 +90,548 @@ which are covered in the lesson on the shell.
...
+> ~~~ {.output}
+> ~~~
+
+
+> ## Adding a help message {.challenge}
+>
+> Separately,
+> modify `readings.py` so that if no parameters are given
+> (i.e., no action is specified and no filenames are given),
+> it prints a message explaining how it should be used.
```
Comment (abostroem) [inline instructors.md]: Can this point to reading-07.py?

File: instructors.md [hunk 5]
```diff
@@ -90,18 +90,548 @@ which are covered in the lesson on the shell.
...
+> ~~~
+
+> ## Adding a default action {.challenge}
+>
+> Separately,
+> modify `readings.py` so that if no action is given
+> it displays the means of the data.
+
+> ~~~ {.output}
+> ~~~
```
Comment (abostroem) [inline instructors.md]: And this point to readings-08.py?

**Comment by abostroem** [comment]
@PBarmby This looks good to me. 2 minor inline comments and then it's ready to be merged. Thank you so much!

**Comment by PBarmby** [comment]
Changes done, ready to go.

**Pull request merged**